automatminer/automl/keras_wrapper.py

```diff
@@ -0,0 +1,157 @@
+        self.estimator = None
+        self.classifier = None
+        self.kfold_splits = kfold_splits
+        if mode in _classifier_modes:
+            self.mode = "classification"
+        else:
+            self.mode = "regression"
+        self.fitted_pipeline_ = None
+
+    def getModel(self):
+        return self.model_
+
+    def fit(self, X, y, **kwargs):
+        self.set_params(**kwargs)
+        model = keras.models.Sequential()
+        first = True
+        if self.input_noise > 0:
+            model.add(keras.layers.GaussianNoise(self.input_noise, input_shape=X.shape[1:]))
+        num_maxout_features = 2
+        dense_kwargs = {"init": self.init}
+        if self.use_maxnorm:
+            dense_kwargs["W_constraint"] = keras.constraints.maxnorm(2)
+
+        # hidden layers
+        for layer_size in range(self.hidden_layer_sizes):
+            if first:
+                if self.use_maxout:
+                    model.add(keras.legacy.layers.MaxoutDense(output_dim=self.hidden_layer_sizes / num_maxout_features,
+                                                                input_dim=X.shape[1], init=dense_kwargs["init"],
+                                                                nb_feature=num_maxout_features))
+                else:
+                    model.add(keras.layers.Dense(units=self.units, input_dim=X.shape[1], kernel_initializer=dense_kwargs["init"]))
+                    model.add(keras.layers.Activation(self.activation))
+                    first = False
+            else:
+                if self.use_maxout:
+                    model.add(keras.legacy.layers.MaxoutDense(output_dim=self.hidden_layer_sizes / num_maxout_features,
+                                                                init=dense_kwargs["init"],
+                                                                nb_feature=num_maxout_features))
+                else:
+                    model.add(keras.layers.Dense(units=self.units, kernel_initializer=dense_kwargs["init"]))
+                    model.add(keras.layers.Activation(self.activation))
+            model.add(keras.layers.Dropout(self.dropout))
+
+        if first:
+            model.add(keras.layers.Dense(output_dim=1, input_dim=X.shape[1], **dense_kwargs))
+        else:
+            model.add(keras.layers.Dense(units=1, kernel_initializer=dense_kwargs["init"]))
+        model.add(keras.layers.Activation(self.activation))
+
+        if self.mode == "classification":
+            model.compile(loss="binary_crossentropy", optimizer=self.optimizer, metrics=["accuracy"])
+        else:
+            model.compile(loss="mse", optimizer=self.optimizer, metrics=["mse"])
+
+        # batches as per configuration
+        for num_iterations, batch_size in self.batch_spec:
+            callbacks = None
+            validation_split = 0.0
+            if self.stop_early and batch_size > 0:
+                callbacks = [EarlyStopping(monitor='val_loss', patience=20, verbose=1)]
```

Comment:
Ur missing an import statement for this early stoppage. It will throw an importerror if it ever gets to this point.